### PR TITLE
fix: fix application info optional fields

### PR DESCRIPTION
**Pull request by Azn9**
**Description:** Change required by https://github.com/Discord4J/discord-json/pull/185 to fix https://github.com/Discord4J/Discord4J/issues/1243

### CODE REVIEW

**Comment by Doc94** [comment]
Currently broken?

File: core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java [hunk 1]
```diff
@@ -356,7 +356,7 @@ public List<String> getRedirectUris() {
      * @return An {@link Optional} containing the interactions endpoint URL for the app if present
      */
     public Optional<String> getInteractionsEndpointUrl() {
-        return data.interactionsEndpointUrl().toOptional();
+        return data.interactionsEndpointUrl().toOptional().flatMap(o -> o);
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java:359]:
```suggestion
        return Possible.flatOpt(data.interactionsEndpointUrl());
```

File: core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java [hunk 2]
```diff
@@ -365,7 +365,7 @@ public Optional<String> getInteractionsEndpointUrl() {
      * @return An {@link Optional} containing the role connection verification URL for the app
      */
     public Optional<String> getRoleConnectionsVerificationUrl() {
-        return data.roleConnectionsVerificationUrl().toOptional();
+        return data.roleConnectionsVerificationUrl().toOptional().flatMap(o -> o);
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java:368]:
```suggestion
        return Possible.flatOpt(data.roleConnectionsVerificationUrl());
```

**Comment by Azn9** [comment]
I forgot this method existed thx @Doc94 x)

**Pull request merged**